subsys/bluetooth/host/hci_core.c

```diff
@@ -2195,6 +2195,25 @@ static void hci_hardware_error(struct net_buf *buf)
 }
 
 #if defined(CONFIG_BT_SMP)
+static bt_conn_ltk_request_hook_t ltk_request_hook;
+
+int bt_conn_ltk_request_hook_install(bt_conn_ltk_request_hook_t cb)
+{
+	bool ok;
```

Comment:
I'm somewhat against this. I'm worried that would hide coding errors. I suggest calling install at boot, i.e. in a SYS_INIT. Would that work for you?

I'm also considering a weak function instead of the install hook. That way single-definition is enforced at link-time. I think it also saves RAM, ROM and code since our compiler is not smart enough to convert an install-function to a direct call. What do you think of that?

Reply:
Yes, we currently use the `SYS_INIT` macro for our use case.

The weak function approach is fine for the current use case, even though @Marek highlighted an issue once there is a need to merge two use cases that require the hook API in a single application. Currently, we don't have any need to do that but this is a possible future limitation.


However, I have one suggestion regarding the generalization of the Hook API that could help us install hooks for other HCI opcodes if there is a need for that in the future. What do you think about changing the existing API as follows?

```int bt_conn_ltk_request_hook_install(bt_conn_ltk_request_hook_t cb);```
->
```int bt_hci_evt_hook_install(uint16_t evt_opcode,  bt_hci_evt_hook_t cb);```

Then we can define the Kconfig for enabling hooks:

```config BT_HCI_EVT_HOOK```

and implement only one hook instance for handling LTKs in this PR:

```config BT_HCI_EVT_HOOK_LE_LTK_REQUEST```

If the given HCI event opcode is not enabled via the Kconfig, the `bt_hci_evt_hook_install` should reject the installation attempt.

This approach allows us to easily extend the hook functionality for future HCI event opcodes. It also provides a granular way of enabling only the necessary hooks for the application use case (memory savings).